libs/base/Data/List.idr

```diff
@@ -36,3 +36,17 @@ isElem x (y :: xs) with (decEq x y)
                Elem x' (y' :: xs') -> Void
         mkNo f g Here = f Refl
         mkNo f g (There x) = g x
+
+intersectBy : (a -> a -> Bool) -> List a -> List a -> List a
```

Comment:
Why doesn't intersectBy get documentation? And why are there specific cases for []?